libraries/aws/shadow/src/shadow.c

```diff
@@ -0,0 +1,550 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file shadow.c
+ * @brief Implements the user-facing functions of the Shadow library.
+ */
+
+/* Standard includes. */
+#include <stdbool.h>
+#include <string.h>
+
+/* Shadow includes. */
+#include "shadow.h"
+
+/**
+ * @brief The string representing "/shadow/update/accepted".
+ */
+
+/* MISRA Rule 5.4 flags the following macro's name as ambiguous from the
```

Comment:
Same comment about MISRA as before in .h file.

Reply:
This is following the pattern from HTTP
https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/development/libraries/standard/http/src/private/http_client_internal.h#L72

Would appreciate to know if we have a better way to handle this case.
Not sure if we should

Reply:
fixed it in commit [8da68c5](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1081/commits/8da68c5f3ccc011a9ab8f58349cf6273c42ec19c)

shorten the macros and changed XX_OPERATION_XX to XX_OP_XX to avoid 5.4, comments for the macro explains that the macro is used for operations


resolving this comment

feel free to let me know if it needs further changes, thank you